cpp/src/CMakeLists.txt

```diff
@@ -100,3 +100,13 @@ target_include_directories(kim-loader PUBLIC
   ${CMAKE_BINARY_DIR}
   )
 target_link_libraries(kim-loader ${CMAKE_DL_LIBS})
+
+if(${CMAKE_SYSTEM_NAME} STREQUAL CYGWIN)
+  # The KIM API code makes use of non-standard POSIX functions such as mkstemp(), which are 
```

Comment:
Do you have a list of the functions that are non-standard POSIX?  I would prefer to explore a switch to POSIX compliant code, if possible.